controller/src/main/java/io/pravega/controller/server/SegmentHelper.java

```diff
@@ -1000,22 +1000,11 @@ public void noSuchSegment(WireCommands.NoSuchSegment noSuchSegment) {
             public void tableRead(WireCommands.TableRead tableRead) {
                 log.info(requestId, "readTable {} successful.", qualifiedName);
                 List<TableEntry<byte[], byte[]>> tableEntries = tableRead.getEntries().getEntries().stream()
-                                                                         .map(e -> {
-                                                                             WireCommands.TableKey k = e.getKey();
-                                                                             TableKey<byte[]> tableKey =
-                                                                                     new TableKeyImpl<>(getArray(k.getData()),
-                                                                                                        new KeyVersionImpl(k.getKeyVersion()));
-                                                                             return new TableEntryImpl<>(tableKey, getArray(e.getValue().getData()));
-                                                                         }).collect(Collectors.toList());
+                                                                         .map(e -> new TableEntryImpl<>(convertFromWireCommand(e.getKey()), getArray(e.getValue().getData())))
+                                                                         .collect(Collectors.toList());
                 result.complete(tableEntries);
             }
 
-            @Override
```

Comment:
why are you removing this? table key does not exist is an expected response that should be handled. right?

Reply:
`io.pravega.shared.protocol.netty.FailingReplyProcessor#tableKeyDoesNotExist`  call back is invoked when `WireCommands.TableKeyDoesNotExist` is received by the client. This Wirecommand is sent by the PRP(PravegaRequestProcessor) incase `io.pravega.segmentstore.contracts.tables.KeyNotExistsException` is thrown by the TableStore apis. 
Now, `io.pravega.segmentstore.contracts.tables.TableStore#get` does not throw `KeyNotExistsException` incase the key does not exist, it returns a null. The PRP now returns an Entry where the KeyVersion corresponds to `TableKey.NOT_EXISTS` if the key does not exist.

Reply:
it is still a good idea to keep that response available, in case something changes on segment store in future. because that can be a legitimate reponse to read queries.. right?

Also, can we have the same issue with readTableEntries?

Reply:
The default implementation for `ReplyProcessor#tableKeyDoesNotExist` is to throw an `IllegalStateException`.
Updated the code to remove redundant implementation of `tableKeyDoesNotExist` in `readTableKeys(..)` and `readTableEntries(..)`.